### PR TITLE
Improve creation and deletion of the service bindings and the usages

### LINKED ISSUE #195: Improve creation and deletion of the service bindings and the usages
**Description**

- Create Credentials button should be called Request Credentials
  - on click, if no schema for binding is provided, the service binding must be created 
  - on click, if schema for binding is provided we display modal with schema so user can fill in and create binding
create-credentials-modal
  - always after creation screen should refresh and new item on list appears
  - help icon in modal should display info box with information what does it mean to request credentials, what is actually happening here
help
  - confirm info should have header `Credentials requested`
credentials-create-confirmation
- Bind Application button on click shows modal that does sbu and sb
  **1st section of modal called Bind Application, and then we have ruler**
  - help icon in modal should display info box with information what does it mean to create credentials, what is actually happening here
help
  - the first textbox, realtime displays list of all resources of all available types that match the typed in text. 
  - below the textbox you have link called `Set environment prefix` and on click you get another text box where you can set the prefix. Validate length if there are some limits predefined
sbu-prefix
  **2nd section of modal called Create Credentials**
  - display binding parameters if such provided
  - display scroll bar in case scrolling is needed
  - instead of `Select an existing Binding` have a link `select existing credentials`
  - once above link is clicked display dropdown where user can select existing ServiceBindings. And hide schema parameters (if provided) for new binding
sbu-existing-binding
  - we need some help text that explains credentials section like the bind apps section
help
- in case mentioned above links for env prefix and existing bindings, it should be possible to cancel when someone selects them
- any UI text must be consulted with @kazydek 
- Delete Service Binding 
  - Should warn what SBUs will also be deleted/affected. Depends on completion of #1069
sbu-remove
- Delete Service Binding Usage should warn like previously `Warning: Removing Service Binding Usage will prevent your application from accessing the instance after its restart.`

**Reasons**

As described in linked epic.  Also current UX issue with binding is that it is overcomplicated, for example:
- you have option to provide names for sb and sbu even though they have no meaning and can be generated
- to select application you want to bind to, but need to first know that kind of app is it, function/deployment?
- reusing binding was promoted to much
- setting proxy for sbu name was not possible
- status of sbu and sb was not visible
- schema for binding not supported

### PR DESCRIPTION
**Pull request by akucharska**
**Description**

Changes proposed in this pull request:

- add create credentials modal
- update bind application modal
- add delete credentials modal

**Related issue(s)**
Resolves #195

### CODE REVIEW

File: instances/src/components/ServiceInstanceDetails/ServiceInstanceBindings/ServiceInstanceBindings.component.js [hunk 1]
```diff
@@ -74,6 +75,7 @@ class ServiceInstanceBindings extends React.Component {
       deleteBindingUsage,
       serviceInstanceRefetch,
       serviceInstance,
+      callback,
```
Comment (magicmatatjahu) [inline instances/src/components/ServiceInstanceDetails/ServiceInstanceBindings/ServiceInstanceBindings.component.js]: Please change this name to another.

File: instances/src/components/ServiceInstanceDetails/ServiceInstanceBindings/ServiceInstanceBindings.component.js [hunk 2]
```diff
@@ -160,7 +174,11 @@ class ServiceInstanceBindings extends React.Component {
             const secret = el && el.secret;
             return secret && Object.keys(secret).length ? (
               <SecretDataModal
-                title={`Secret "${secret.name}"`}
+                title={
+                  <Fragment>
+                    Secret <Bold>"{secret.name}"</Bold>
```
Comment (magicmatatjahu) [inline instances/src/components/ServiceInstanceDetails/ServiceInstanceBindings/ServiceInstanceBindings.component.js]: Remove semicolons

File: instances/src/components/ServiceInstanceDetails/ServiceInstanceBindings/InfoButton/variables.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export const bindingVariables = {
+  setEnvPrefix:
+    'Defines the prefix of environment variables environment variables that the ServiceBindingUsage injects. The prefixing is disabled by default.',
```
Comment (kazydek) [inline instances/src/components/ServiceInstanceDetails/ServiceInstanceBindings/InfoButton/variables.js]:
```suggestion
    'Defines prefixes of environment variables that the ServiceBindingUsage injects. Prefixing is disabled by default.',
```

File: instances/src/components/ServiceInstanceDetails/ServiceInstanceBindings/InfoButton/variables.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+  serviceBinding:
+    'To use ServiceInstance, you need credentials for this service. To obtain credentials, proceed with this form. One instance can have numerous credentials to use in the Deployment or Function. When you raise a credentials request, the system returns the credentials in the form of a Secret. The system creates a Secret in a given Environment.',
```
Comment (kazydek) [inline instances/src/components/ServiceInstanceDetails/ServiceInstanceBindings/InfoButton/variables.js]:
```suggestion
    'To use the ServiceInstance, you need credentials for the service. Fill in this form to obtain them. One ServiceInstance can have numerous credentials to use in a Deployment or Function. When you raise a request for credentials, the system returns them in the form of a Secret that it creates in a given Environment.',
```

File: instances/src/components/ServiceInstanceDetails/ServiceInstanceBindings/InfoButton/variables.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+  serviceBingingUsage:
+    "The Secret allows you to run the service successfully. However, a problem appears each time you need to change the definition of the yaml file in the Deployment to specify the Secrets' usage. The manual process of editing the file is tedious and time-consuming. Kyma handles it by offering a custom resource called ServiceBindingUsage. This custom resource applies the Kubernetes PodPreset resource and allows you to enforce an automated flow in which the ServiceBindingUsage controller injects credentials into a given Application or Function.",
```
Comment (kazydek) [inline instances/src/components/ServiceInstanceDetails/ServiceInstanceBindings/InfoButton/variables.js]:
```suggestion
    "The Secret allows you to run the service successfully. However, a problem appears each time you need to change the definition of the yaml file in the Deployment to specify the Secrets' usage. The manual process of editing the file is tedious and time-consuming. Kyma handles it by offering a custom resource called ServiceBindingUsage. This custom resource applies the Kubernetes PodPreset resource and allows you to enforce an automated flow in which the ServiceBindingUsage controller injects credentials into a given Deployment or Function.",
```

**Pull request merged**